pkg/catalog/catalog.go

```diff
@@ -1711,6 +1719,99 @@ func (c *Catalog) PrepareExpiredCommits(ctx context.Context, repositoryID string
 	return c.Store.SaveGarbageCollectionCommits(ctx, repository, previousRunID)
 }
 
+func createParquetFile(entries []DBEntry) (string, error) {
```

Comment:
I would want this to return a parquet reader. Together with the iterator approach suggested by Itai in another comment - we could stream the entries from the KV store to the final Parquet file without saving it in memory or in the file system.